src/runtime/hexagon_remote/halide_remote.cpp

```diff
@@ -157,19 +168,37 @@ int halide_hexagon_remote_set_performance(
         return -1;
     }
 
-    request.type = HAP_power_set_mips_bw;
-    request.mips_bw.set_mips        = set_mips;
-    request.mips_bw.mipsPerThread   = mipsPerThread;
-    request.mips_bw.mipsTotal       = mipsTotal;
-    request.mips_bw.set_bus_bw      = set_bus_bw;
-    request.mips_bw.bwBytePerSec    = ((uint64_t) bwMegabytesPerSec) << 20;
-    request.mips_bw.busbwUsagePercentage = busbwUsagePercentage;
-    request.mips_bw.set_latency     = set_latency;
-    request.mips_bw.latency         = latency;
-    retval = HAP_power_set(NULL, &request);
-    if (0 != retval) {
-        log_printf("HAP_power_set(HAP_power_set_mips_bw) failed (%d)\n", retval);
-        return -1;
+    if (use_dlopenbuf()) {
```

Comment:
At best, this seems coincidentally correct (using dlopenbuf as a proxy for the v2 of this being present). Is there some other way to determine if HAP_power_set_DCVS_v2 is available? Maybe it should simply try HAP_power_set with HAP_power_set_DCVS_v2, and if it fails, then use the old one?

Also, are these actually mutually exclusive? It seems like they're setting different things (except latency).

Reply:
I tested the "try old if new fails" but it doesn't set DCVS correctly. using dlopenbuf() to differentiate the two seems the only option for now, although it doesn't seem not clean logically.

Reply:
There *must* to be a way to do this in the HAP API. If there isn't, Qualcomm should add one. At the very least, trying the new one and falling back to the old one really should work. Why doesn't that work?